server/src/main/scala/org/http4s/server/middleware/HSTS.scala

```diff
@@ -0,0 +1,27 @@
+package org.http4s
+package server
+package middleware
+
+import org.http4s.headers.`Strict-Transport-Security`
+import fs2._
+import fs2.interop.cats._
+
+import scala.concurrent.duration.FiniteDuration
+import scala.concurrent.duration._
+
+/** [[Middleware]] to add HTTP Strict Transport Security (HSTS) support adding
+  * the Strict Transport Security headers
+  */
+object HSTS {
+
+  def apply(service: HttpService, maxAge: FiniteDuration = 365.days, includeSubDomains: Boolean = true, preload: Boolean = false): HttpService = Service.lift { req =>
+
+    val header = `Strict-Transport-Security`.unsafeFromDuration(maxAge, includeSubDomains, preload)
```

Comment:
I'm thinking to define several `apply` methods to pass directly the header and one with a default age of 1y. Both cases would make `apply` total and then convert the current `apply` to `unsafeFromDuration`

WDYT?